.github/workflows/assemble.yml

```diff
@@ -17,10 +17,22 @@ jobs:
           java-version: ${{ matrix.java }}
           distribution: temurin
       - name: Setup docker (missing on MacOS)
+        id: setup_docker
         if: runner.os == 'macos'
         uses: douglascamata/setup-docker-macos-action@main
         with:
           upgrade-qemu: true
+          colima:  v0.6.8
       - name: Run Gradle (assemble)
+        if: runner.os == 'macos' && steps.setup_docker.outcome != 'success'
```

Comment:
@peternied could you please validate me here: for MacOS, if the Docker setup step fails, we'll report success nonetheless

Reply:
IMO no reason to use these runners if we are ignoring the results.  Why not remove `macos-13` from the matrix on line 11 or alternatively exclude `macos-13` from the matrix [1].


- [1] https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#excluding-matrix-configurations

Reply:
There is a reason, we introduced these checks exactly because `assembly` was failing on different OSes. We do not ignore results, we only report "success" when the runner was not able to setup Docker, the "real" check won't even start. This is not the only or best option, but the Docker setup fails from time to time, most of the checks are full run.

Reply:
If the mac-runners always fail to setup we will not know that assemble fails on MacOS :/  Then suddenly they might start again and fail on an unrelated change.   I know I won't pay attention to the log files to check and see if its periodically succeeding - @reta are you saying that you'd check in at that level?  If you are willing then I am OK to move forward with this kind of change.